packages/chevrotain/src/parse/parser/traits/recoverable.ts

```diff
@@ -271,9 +275,12 @@ export class Recoverable {
     let nextToken = this.LA(1)
     let k = 2
     while (true) {
-      const nextTokenType: any = nextToken.tokenType
-      if (includes(allPossibleReSyncTokTypes, nextTokenType)) {
-        return nextTokenType
+      const foundMatch = allPossibleReSyncTokTypes.find((resyncTokType) => {
+        const canMatch = tokenMatcher(nextToken, resyncTokType)
+        return canMatch
+      })
+      if (foundMatch !== undefined) {
+        return foundMatch
```

Comment:
I think we can skip the test case in this situation because:
1. we know `tokenMatcher` handles categories.
2. coverage won't be affected.
2. The overhead / effort of adding a test case may be too large for the benefit gained.